src/gt4py/storage/storage.py

```diff
@@ -14,604 +14,1112 @@
-        if len(self.strides) < len(self.layout_map):
-            return False
-        for dim in reversed(np.argsort(self.layout_map)):
-            if self.strides[dim] < stride:
-                return False
-            stride = self.strides[dim]
-
-        # check alignment
-        if (
-            self.ctypes.data + np.sum([o * s for o, s in zip(self.default_origin, self.strides)])
-        ) % gt_backend.from_name(self.backend).storage_info["alignment"]:
-            return False
-
-        return True
-
-    def _finalize_view(self, obj):
-        pass
-
-    def synchronize(self):
-        pass
-
-    def host_to_device(self, force=False):
-        pass
-
-    def device_to_host(self, force=False):
-        pass
-
-    def __iconcat__(self, other):
-        raise NotImplementedError("Concatenation of Storages is not supported")
-
-
-class GPUStorage(Storage):
-    @classmethod
-    def _construct(cls, backend, dtype, default_origin, shape, alignment, layout_map):
-
-        (raw_buffer, field) = storage_utils.allocate_gpu(
-            default_origin, shape, layout_map, dtype, alignment * dtype.itemsize
-        )
-        obj = field.view(_ViewableNdarray)
-        obj = obj.view(GPUStorage)
-        obj._raw_buffer = raw_buffer
-        obj.default_origin = default_origin
-        return obj
+    def axes(self):
+        return "".join([c for m, c in zip(self._mask, "IJK") if m])
 
     @property
-    def __cuda_array_interface__(self):
-        array_interface = self.__array_interface__
-        array_interface["version"] = 2
-        array_interface["strides"] = self.strides
-        array_interface.pop("offset", None)
-        return array_interface
-
-    def _check_data(self):
-        # check that memory of field is within raw_buffer
-        if (
-            not self.ctypes.data >= self._raw_buffer.data.ptr
-            and self.ctypes.data + self.itemsize * (self.size - 1)
-            <= self._raw_buffer[-1:].data.ptr
-        ):
-            raise Exception("The buffers are in an inconsistent state.")
-
-    def copy(self):
-        res = super().copy()
-        res.gpu_view[...] = self.gpu_view
-        cp.cuda.Device(0).synchronize()
-        return res
+    def ndim(self):
+        return sum(self._mask)
 
     @property
-    def gpu_view(self):
-        return storage_utils.gpu_view(self)
-
-    def __setitem__(self, key, value):
-        if hasattr(value, "__cuda_array_interface__"):
-            gpu_view = storage_utils.gpu_view(self)
-            gpu_view[key] = cp.asarray(value)
-            cp.cuda.Device(0).synchronize()
-            return value
-        else:
-            return super().__setitem__(key, value)
+    def default_origin(self):
+        return self._default_origin
 
     @property
-    def _is_clean(self):
-        return True
+    def alignment(self):
+        return self._alignment
 
     @property
-    def _is_host_modified(self):
-        return False
+    def layout_map(self):
```

Comment:
This attribute is not tested. It would be very helpful to add some documentation or type hinting about this attribute.